src/newgrf.cpp

```diff
@@ -3544,49 +3542,41 @@ static ChangeInfoResult IndustriesChangeInfo(uint indid, int numinfo, int prop,
 									grfmsg(2, "IndustriesChangeInfo: Attempt to use industry tile %u with industry id %u, not yet defined. Ignoring.", local_tile_id, indid);
 								} else {
 									/* Declared as been valid, can be used */
-									itt[k].gfx = tempid;
+									it.gfx = tempid;
 								}
-							} else if (itt[k].gfx == 0xFF) {
-								itt[k].ti.x = (int8)GB(itt[k].ti.x, 0, 8);
-								itt[k].ti.y = (int8)GB(itt[k].ti.y, 0, 8);
+							} else if (it.gfx == 0xFF) {
+								it.ti.x = (int8)GB(it.ti.x, 0, 8);
+								it.ti.y = (int8)GB(it.ti.y, 0, 8);
 
 								/* When there were only 256x256 maps, TileIndex was a uint16 and
-								 * itt[k].ti was just a TileIndexDiff that was added to it.
+								 * it.ti was just a TileIndexDiff that was added to it.
 								 * As such negative "x" values were shifted into the "y" position.
 								 *   x = -1, y = 1 -> x = 255, y = 0
 								 * Since GRF version 8 the position is interpreted as pair of independent int8.
 								 * For GRF version < 8 we need to emulate the old shifting behaviour.
 								 */
-								if (_cur.grffile->grf_version < 8 && itt[k].ti.x < 0) itt[k].ti.y += 1;
+								if (_cur.grffile->grf_version < 8 && it.ti.x < 0) it.ti.y += 1;
 							}
 						}
 
-						if (!ValidateIndustryLayout(copy_from, size)) {
+						if (!ValidateIndustryLayout(layout)) {
 							/* The industry layout was not valid, so skip this one. */
 							grfmsg(1, "IndustriesChangeInfo: Invalid industry layout for industry id %u. Ignoring", indid);
 							new_num_layouts--;
 							j--;
 						} else {
-							tile_table[j] = CallocT<IndustryTileTable>(size);
-							memcpy(tile_table[j], copy_from, sizeof(*copy_from) * size);
+							new_layouts.push_back(layout);
 						}
 					}
 				} catch (...) {
-					for (int i = 0; i < new_num_layouts; i++) {
-						free(tile_table[i]);
-					}
-					free(tile_table);
-					free(itt);
 					throw;
 				}
```

Comment:
Should I just go all the way and remove the entire try-catch block here? I'm not even sure what it'd be catching.

Reply:
catch-all block that just rethrows? yeah, kill it